### PR TITLE
DDF-3797 Filter duplicate claim types from SAML assertions

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Ensures the claim types we request from claims handlers are all unique.

#### Who is reviewing it? 
@bakejeyner @blen-desta 

#### Select relevant component teams: 
@codice/security 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@brendan-hofmann 
@stustison

#### How should this be tested? (List steps with links to updated documentation)
1. Install DDF using the minimal profile
2. Login to /admin. In the Web Context Policy Manager, change the auth type for /admin to SAML|BASIC:
```
/admin=SAML|BASIC
```
3. From the Karaf console, turn up the log level for the Login Filter to TRACE:
```
> log:set TRACE org.codice.ddf.security.filter.login
```
4. Logout
5. Log back into /admin. Find the SAML assertion in the log and verify that there are no duplicate Attribute elements in the AttributeStatement, as shown in the image below.

NOTE: Attributes with the same Name and NameFormat but different values are expected.

#### Any background context you want to provide?
The STSRealm creates a list of claims to request from registered claims handlers. It pulls these claims from the Web Context Policy Manager and the STS Client config. If a claim is listed in both, then it will be requested multiple times from the claims handlers, resulting in a bloated assertion.

#### What are the relevant tickets?
[DDF-3797](https://codice.atlassian.net/browse/DDF-3797)

#### Screenshots (if appropriate)
screen shot 2018-04-25 at 1 39 11 pm

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by SmithJosh** [comment]
build now

**Comment by blen-desta** [comment]
Hero 💯

File: platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java [hunk 1]
```diff
@@ -407,21 +407,16 @@ private void setClaimsOnStsClient(STSClient stsClient, Element claimsElement) {
...
   protected Element createClaimsElement() {
     Element claimsElement = null;
-    List<String> claims = new ArrayList<>();
+    Set<String> claims = new LinkedHashSet<>();
     claims.addAll(getClaims());
 
     if (contextPolicyManager != null) {
       Collection<ContextPolicy> contextPolicies = contextPolicyManager.getAllContextPolicies();
-      Set<String> attributes = new LinkedHashSet<>();
       if (contextPolicies != null && contextPolicies.size() > 0) {
```
Comment (SmithJosh) [inline platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java]: ✏️ Collection.isEmpty() should be used to test for emptiness

File: platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java [hunk 2]
```diff
@@ -407,21 +407,16 @@ private void setClaimsOnStsClient(STSClient stsClient, Element claimsElement) {
...
+          claims.addAll(contextPolicy.getAllowedAttributeNames());
         }
       }
-
-      if (attributes.size() > 0) {
-        claims.addAll(attributes);
-      }
     }
 
     if (claims.size() != 0) {
```
Comment (SmithJosh) [inline platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java]: ✏️ Collection.isEmpty() should be used to test for emptiness

**Comment by SmithJosh** [comment]
build now

**Pull request merged**